HISTORY.md

```diff
@@ -1,3 +1,110 @@
+# TileDB v2.17.0 Release Notes
+
+## Disk Format
+
+* Storage format version is now 20, with changes to support enumerated types ([#4051](https://github.com/TileDB-Inc/TileDB/pull/4051)).
+
+## Configuration Changes
+
+* Add Enumeration max size configuration. [#4308](https://github.com/TileDB-Inc/TileDB/pull/4308)
+
+## New features
+
+* Enumerated data types (aka: "categoricals" in Pandas, "factors" in R). [#4051](https://github.com/TileDB-Inc/TileDB/pull/4051)
+* Support optimized creation of set membership (`IN`, `NOT_IN`) QueryConditions. [#4164](https://github.com/TileDB-Inc/TileDB/pull/4164). Replaces create+combine pattern for specifying the target set.
+* Add `tiledb::as_built` namespace to reflect build-time configuration. [#4112](https://github.com/TileDB-Inc/TileDB/pull/4112)
```

Comment:
Nit:  It is `tiledb::as_built` in the implementation, but the (uglier ?) `tiledb::AsBuilt` in the interface.  Should the notes reflect this?